### PR TITLE
Cleanup - `disable_bpf_loader_instructions`

### PR DESCRIPTION
**Pull request by Lichtso**
#### Problem
The feature was forced in v1.17.20 on MNB.

#### Summary of Changes
Removes it from master and v1.18 as well to stay in sync.

Feature Gate Issue: #34424

### CODE REVIEW

**Comment by HaoranYi** [comment]
@Lichtso I pushed a few commits to update the test and bench. CI passed. I think this PR is ready. Can you take a look?

File: runtime/src/bank_client.rs [hunk 1]
```diff
@@ -331,7 +331,7 @@ impl BankClient {
         self.bank.set_sysvar_for_tests(sysvar);
     }
 
-    #[cfg(feature = "dev-context-only-utils")]
+    /// Only used for tests and benchmarks.
     pub fn advance_slot(
```
Comment (alessandrod) [inline runtime/src/bank_client.rs:335]: This seems wrong? Why this change?

File: runtime/src/loader_utils.rs [hunk 2]
```diff
@@ -206,6 +212,48 @@ pub fn load_upgradeable_program(
...
+        mint_keypair,
+        &buffer_keypair,
+        &program_keypair,
+        authority_keypair,
+        name,
+    );
+    program_keypair.pubkey()
+}
+
+pub fn load_upgradeable_program_and_advance_slot(
```
Comment (alessandrod) [inline runtime/src/loader_utils.rs:234]: Instead of removing the dev-context-only-utils from other places, I think it
needs to be added here? And possibly in other places too

File: runtime/src/bank_client.rs (hunk 1 above)
Comment (HaoranYi) [reply runtime/src/bank_client.rs:335]: This fn is now used in benches. Benches are not built in dev-context.  If it is decorated with dev-context-only, then benches build fails.

File: runtime/src/loader_utils.rs (hunk 2 above)
Comment (HaoranYi) [reply runtime/src/loader_utils.rs:234]: See comments above.

File: runtime/src/bank_client.rs (hunk 1 above)
Comment (HaoranYi) [reply runtime/src/bank_client.rs:335]: credit to @pgarg66 
We can update the cargo.toml of bench to include dev-context-util dependency and keep the attribute.

File: runtime/src/loader_utils.rs (hunk 2 above)
Comment (HaoranYi) [reply runtime/src/loader_utils.rs:234]: Yes. Added.

**Review by alessandrod — APPROVED** [review]
